### PR TITLE
Add context.Context to relevant API

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: neo4j/internal/connector/connector.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (c Connector) Connect(address string, boltLogger log.BoltLogger) (db.Connec
 	}
 	config := tls.Config{InsecureSkipVerify: c.SkipVerify, RootCAs: c.RootCAs, ServerName: serverName}
 	tlsconn := tls.Client(conn, &config)
-	err = tlsconn.Handshake()
+	err = tlsconn.HandshakeContext(ctx)
```
Comment (fbiville) [inline neo4j/internal/connector/connector.go:89]: this is where I'm not sure... should I use another context param for this specific TLS handshake or not?

**Review by fbiville — COMMENTED** [review]
Another open question is: do I need to expose a `context.Context` on the low-level chunker and dechunker since they write/read data and could possibly take a long time?
There is no direct support for this in Go's I/O package but we can set read/write deadline on the TCP connection. Creating a bridge API with `context.Context` which under the hood translate read/write deadline should be straightforward.

File: testkit-backend/backend.go [hunk 2]
```diff
@@ -608,7 +606,6 @@ func (b *backend) handleRequest(req map[string]interface{}) {
 				"Optimization:ConnectionReuse",
 				"Optimization:ImplicitDefaultArguments",
 				"Optimization:PullPipelining",
-				"Temporary:ConnectionAcquisitionTimeout",
```
Comment (fbiville) [inline testkit-backend/backend.go:611]: @robsdedude this PR unfortunately conflicts with timeout configuration on the testkit backend side
if this PR is merged, timeout/deadlines/... would be configured through each API via `context.Context` object, so it won't be possible to set a timeout before calling the related operation

File: neo4j/test-integration/examples_test.go [hunk 3]
```diff
@@ -540,21 +518,11 @@ func createDriverWithCustomizedConnectionPool(uri, username, password string) (n
 	return neo4j.NewDriver(uri, neo4j.BasicAuth(username, password, ""), func(config *neo4j.Config) {
 		config.MaxConnectionLifetime = 30 * time.Minute
 		config.MaxConnectionPoolSize = 50
-		config.ConnectionAcquisitionTimeout = 2 * time.Minute
 	})
 }
 
 // end::config-connection-pool[]
 
-// tag::config-connection-timeout[]
```
Comment (fbiville) [inline neo4j/test-integration/examples_test.go:549]: not sure if that's OK to just remove the entire tag

File: testkit-backend/backend.go (hunk 2 above)
Comment (robsdedude) [reply testkit-backend/backend.go:611]: Couldn't you store the timeout that TestKit requires somewhere in you TestKit back end to then later pass it into a function call?

File: neo4j/test-integration/examples_test.go (hunk 3 above)
Comment (robsdedude) [reply neo4j/test-integration/examples_test.go:549]: If the driver manual tries to pull this in, you'd also have to adjust that (remove the reference from there).

File: testkit-backend/backend.go (hunk 2 above)
Comment (fbiville) [reply testkit-backend/backend.go:611]: actually, you're right, it should work

**Pull request closed without merging**